ui/src/shared/utils/filterUnusedVars.ts

```diff
@@ -48,6 +48,19 @@ export const createdUsedVarsCache = (variables: Variable[]) => {
   }, {})
 }
 
+export const filterUnusedVarsBasedOnQuery = (
+  variables: Variable[],
+  queryTexts: string[]
+) => {
+  const varsInUse = variables.filter(variable =>
+    queryTexts.some(text => isInQuery(text, variable))
```

Comment:
ugh does this take into account nested variables? nested variables are the worst things!

Reply:
`isInQuery` does not, but that's why we added this :)

https://github.com/influxdata/influxdb/pull/19010/files/53904bedecb318e191d36c25903d703881ea491a#diff-8e45b03aeca264900ab0daa6f18aacd0R61